Sources/SpeziStudyDefinition/StudyBundle.swift

```diff
@@ -0,0 +1,319 @@
+//
+// This source file is part of the My Heart Counts iOS application based on the Stanford Spezi Template Application project
+//
+// SPDX-FileCopyrightText: 2025 Stanford University
+//
+// SPDX-License-Identifier: MIT
+//
+
+// swiftlint:disable all
+
+import Foundation
+import class ModelsR4.Questionnaire
+import SpeziFoundation
+import UniformTypeIdentifiers
+import OSLog
+
+
+extension UTType {
+    public static let speziStudyBundle = UTType(filenameExtension: "spezistudybundle", conformingTo: .package)!
+}
+
+
+/// A handle for working with a Study Definition bundle
+public struct StudyBundle: Hashable, Identifiable, Sendable {
+    public static let fileExtension = "spezistudybundle"
+    
+    private static let logger = Logger(subsystem: "edu.stanford.SpeziStudy", category: "StudyBundle")
+    
+    /// The file url of the study definition bundle.
+    public let bundleUrl: URL
+    public let studyDefinition: StudyDefinition // TODO what if the study bundle gets written to after we've read the StudyDef?
```

Comment:
The best way would probably be to have a file listener or something similar? Maybe a future feature that can be enabled and disabled? Probably not very relevant to get started. Might be a good GitHub issue to tackle.

Reply:
i was thinking the opposite, actually. ie, that we somehow should make it illegal to mutate the on-disk file. i don't think there's any scenario where we'd actually want that behavior (ie, where we'd want to allow any part of the application (including the StudyBundle itself) to mutate the on-disk study).

we could implement this using a file listener, but then we'd also need to turn the StudyBundle into a reference type.

Reply:
Fair point, maybe we put out an error log if a file has changed? It will still be able to re-load a study manually anyways if things changed and we could make that work and point that out in the error logging?

Reply:
Sounds good; I like the documentation and warning approach for now.

Agree that this monitoring approach would be great in the future but not needed now. I would also assume that the safest approach would be to copy a bundle in a dedicated folder on loading and only access it there while the original files could change but won't be considered until it is loaded again which means an other copy and the internal re-loading?